### ISSUE TITLE
Autolink doesn't handle well urls containing emails as their part

### ISSUE THREAD

**Issue by imankulov** [Suor/autolink#2]
How to reproduce:

```
In [1]: print autolink.linkify('http://example.com?email=foo@example.com')

<a href="http://example.<a href="mailto:com?email=foo@example.com">com?email=foo@example.com</a>">http://example.<a href="mailto:com?email=foo@example.com">com?email=foo@example.com</a></a>
```

**Comment by Suor** [comment]
Published in 0.1.2

**Comment by imankulov** [comment]
Thanks a lot, @Suor !
